packages/esm-patient-search-app/src/patient-search/patient-search.component.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { useEffect, useCallback } from 'react';
+import debounce from 'lodash-es/debounce';
+import isEmpty from 'lodash-es/isEmpty';
+import Link from 'carbon-components-react/es/components/Link';
+import PaginationNav from 'carbon-components-react/es/components/PaginationNav';
+import PatientSearchResults from '../patient-search-result/patient-search-result.component';
+import EmptyDataIllustration from './empty-data-illustration.component';
+import { Tile } from 'carbon-components-react/es/components/Tile';
+import { useTranslation } from 'react-i18next';
+import { performPatientSearch } from './patient-search.resource';
+import styles from './patient-search.scss';
+import Search from 'carbon-components-react/es/components/Search';
+import { SearchedPatient } from '../types/index';
+
+interface PatientSearchProps {
+  hidePanel?: () => void;
+}
+
+const PatientSearch: React.FC<PatientSearchProps> = ({ hidePanel }) => {
+  const customReprestation =
+    'custom:(patientId,uuid,identifiers,display,' +
+    'patientIdentifier:(uuid,identifier),' +
+    'person:(gender,age,birthdate,birthdateEstimated,personName,display),' +
+    'attributes:(value,attributeType:(name)))';
+  const searchTimeout = 300;
+  const resultsPerPage = 5;
+
+  const [searchTerm, setSearchTerm] = React.useState('');
+  const [emptyResult, setEmptyResult] = React.useState(false);
+  const [searchResults, setSearchResults] = React.useState<Array<SearchedPatient>>([]);
+  const [pagedResults, setPagedResults] = React.useState<Array<SearchedPatient>>([]);
```

Comment:
For the pagination we have a hook that we used in patient-chart.

Let's see if the hook can be (or is already) moved into esm-core (into esm-react-utils to be more specific).

Reply:
Moved this hook `usePagination` from [@openmrs/esm-common-lib](https://github.com/openmrs/openmrs-esm-patient-chart/tree/master/packages/esm-patient-common-lib) to `esm-react-utils`

Reply:
Awesome - thanks!